Sources/Host/Darwin/Platform.cpp

```diff
@@ -118,6 +118,26 @@ std::string Platform::GetThreadName(ProcessId pid, ThreadId tid) {
 const char *Platform::GetSelfExecutablePath() {
   return LibProc::GetExecutablePath(getpid());
 }
+
+ErrorCode Platform::TranslateKernError(kern_return_t kret) {
+  switch (kret) {
+  case KERN_SUCCESS:
+    return ds2::kSuccess;
+  case KERN_FAILURE:
+    return ds2::kErrorUnknown;
+  case KERN_MEMORY_FAILURE:
+  case KERN_INVALID_ADDRESS:
+    return ds2::kErrorInvalidAddress;
+  case KERN_PROTECTION_FAILURE:
+  case KERN_NO_ACCESS:
+    return ds2::kErrorNoPermission;
+  case KERN_MEMORY_ERROR:
```

Comment:
Same for `KERN_MEMORY_ERROR`. `kErrorNoMemory` is only good for `KERN_RESOUCE_SHORTAGE` I think.